client/types/callbacks.go

```diff
@@ -45,18 +45,18 @@ type Callbacks interface {
 	// May be called after Start() is called and every time a connection is established to the Server.
 	// For WebSocket clients this is called after the handshake is completed without any error.
 	// For HTTP clients this is called for any request if the response status is OK.
-	OnConnect()
+	OnConnect(ctx context.Context)
```

Comment:
Can we add a comment (one for the entire interface) that explains how the callback methods are supposed to treat the context? I think we expect them to honour the cancellation.

Reply:
yep. Can do!